core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/AbstractGeometricStuckCriterion.java

```diff
@@ -18,29 +18,28 @@
  * 
  * @param <Solution_> the solution type
  */
-public abstract class AbstractGeometricRestartStrategy<Solution_> implements RestartStrategy<Solution_> {
+public abstract class AbstractGeometricStuckCriterion<Solution_> implements StuckCriterion<Solution_> {
+    protected static final Logger logger = LoggerFactory.getLogger(AbstractGeometricStuckCriterion.class);
     private static final double GEOMETRIC_FACTOR = 1.4; // Value extracted from the cited paper
-    protected final Clock clock;
-    protected final Logger logger = LoggerFactory.getLogger(AbstractGeometricRestartStrategy.class);
-    protected final double scalingFactor;
+    private static final long GRACE_PERIOD_MILLIS = 30_000; // Same value as DiminishedReturnsTermination
```

Comment:
Questionable. If the diminished termination is enabled, don't we want the solver to be able to restart before the termination triggers? Then arguably this should be _less_ than whatever the termination is configured to be. But I'm not sure if we even want that.

Should we detect the termination in the solver config? Should we depend on whatever values the user has configured there? I'm not sure either.

Reply:
I understand your viewpoint and struggled to select an appropriate default value. That's why I opted for the same value as the diminished termination.

However, I believe we should not rely on that termination at all. This should stand as an independent value.